src/pages/Search.vue

```diff
@@ -274,46 +273,54 @@ export default {
-          components: { AssetCard },
-          router: this.$router,
-          store: this.$store,
-          render: h => h(AssetCard, {
-            props: { asset },
-            attrs: { id: popupId },
+        let marker = get(window.stlMapMarkers, assetId)
+
+        if (!marker) {
+          const popupId = `map-popup-${assetId}`
+          const popup = new mapboxgl.Popup({
+            closeButton: false,
+            className: 'stl-map-search-popup'
+          }).setHTML(`<div id="${popupId}"></div>`)
+
+          // Use render function to spare vue compiler
+          const PopupContent = Vue.extend({
+            components: { AssetCard },
+            router: this.$router,
+            store: this.$store,
+            render: h => h(AssetCard, {
+              props: { asset },
+              attrs: { id: popupId },
+            })
           })
-        })
 
-        const marker = new mapboxgl.Marker(el)
-          .setLngLat(f.geometry.coordinates)
-          .addTo(this.map)
-          .setPopup(popup.setHTML(`<div id="${popupId}"></div>`))
+          marker = new mapboxgl.Marker(el)
+            .setLngLat(f.geometry.coordinates)
+            .addTo(this.map)
+            .setPopup(popup)
 
-        window.stlMapMarkers[assetId] = marker
+          set(window.stlMapMarkers, assetId, marker)
 
-        popup.on('open', () => {
-          // Mount PopupContent once
-          const isPopupEmpty = !this.populatedMapMarkers[assetId]
+          popup.on('open', async () => {
+            // Mount PopupContent once
+            const isPopupEmpty = !this.populatedMapMarkers[assetId]
 
-          if (isPopupEmpty) {
-            // can only mount Vue component once mapbox injects popup in DOM
-            new PopupContent().$mount(`#${popupId}`)
-            this.populatedMapMarkers[assetId] = document.getElementById(markerId)
-          }
+            if (isPopupEmpty) {
+              // can only mount Vue component once mapbox injects popup in DOM
+              new PopupContent().$mount(`#${popupId}`)
+              this.populatedMapMarkers[assetId] = document.getElementById(markerId)
+              // Hack to force appropriate initial positionning
+              marker.togglePopup()
+              marker.togglePopup()
+            }
 
-          this.populatedMapMarkers[assetId].style.display = 'none'
-        })
-        popup.on('close', () => {
-          // all popups (even empty ones) get closed when markers are destroyed
-          if (this.populatedMapMarkers[assetId]) this.populatedMapMarkers[assetId].style.display = ''
-        })
-      })
+            this.populatedMapMarkers[assetId].style.display = 'none'
+          })
+          popup.on('close', () => {
+            // all popups (even empty ones) get closed when markers are destroyed
+            if (this.populatedMapMarkers[assetId]) this.populatedMapMarkers[assetId].style.display = ''
+          })
+        }
+      }, { concurrency: 2 }) // pMap
```

Comment:
Thanks for porting. I would mention it was ported from heroes-platform-demo, which might help debugging in the future.